india_compliance/gst_india/doctype/purchase_reconciliation_tool/exporter.py

```diff
@@ -0,0 +1,409 @@
+                ws.cell(row=i + 3, column=j + 1).number_format = "0.00"
+                ws.column_dimensions[column].width = width
+                ws.row_dimensions[1].height = 30
+                ws.row_dimensions[2].height = 30
+        return ws
+
+    def save_workbook(self, wb, file_name):
+        wb.save(file_name)
+        return wb
+
+    def read_workbook(self, file_name):
+        file_path = frappe.get_site_path("private", "files", file_name)
+        wb = openpyxl.load_workbook(file_path, read_only=False, keep_vba=True)
+        return wb
+
+    def add_worksheet(
+        self,
+        data,
+        sheet_name,
+        position,
+        sub_header_color=None,
+        body_color="FFFFFF",
+        font_size=9,
+        row=1,
+        start_column=1,
+        increment_by=1,
+        save=False,
+        workbook=None,
+        file_name=None,
+    ):
+        if not workbook:
+            workbook = self.create_workbook()
+
+        worksheet = self.create_worksheet(workbook, sheet_name, position)
+        self.append_data(
+            worksheet, data, add_row=row, add_column=start_column, is_common_header=True
+        )
+
+        # self.update_header_data(worksheet, header, row=row)
+        # self.setup_header_for_invoice_data(worksheet, range=range, color=header_color)
+        # self.update_data(worksheet, data, header, color=body_color)
+
+        if save:
+            self.save_workbook(workbook, file_name)
+
+        return workbook
+
+
+@frappe.whitelist()
+def export_data_to_xlsx(args, column_widths=None):
+    if isinstance(args, str):
+        args = frappe.parse_json(args)
+
+    exporter = ExcelExporter()
+    workbook = exporter.create_workbook()
+
+    for i, sheet_name in enumerate(args.get("sheet_names")):
+        worksheet = exporter.create_worksheet(workbook, sheet_name, i)
+        exporter.append_data(
+            worksheet,
+            data=args.get("common_header"),
+            add_row=1,
+            add_column=1,
+            is_common_header=True,
+            wrap_text=False,
+            horizontal="left",
+        )
+
+        # 0th index of args array contains the header data
+        if sheet_name == "Summary Data" and args.get("match_summary"):
+            data = args.match_summary[1:]
+            data_header = [args.match_summary[0]]
+            column_widths = [20, 20, 12, 18, 18, 12]
+            header_color = "d9d9d9"
+            body_color = "f2f2f2"
+            add_row = 5
+            additional_style = [
+                {
+                    "min_row": 6,
+                    "min_col": 4,
+                    "max_col": 5,
+                    "bg_color": "f2dcdb",
+                }
+            ]
+
+        if sheet_name == "Supplier Data" and args.get("supplier_summary"):
+            data = args.supplier_summary[1:]
+            data_header = [args.supplier_summary[0]]
+            column_widths = [20, 20, 12, 12, 18, 18, 12]
```

Comment:
After you have JS in backend, all properties related to column should go together. Name, width, style, etc. It would make more sense and improve control.